### PR TITLE
Big geocodings cause statement timeouts

### LINKED ISSUE #3143: Big geocodings cause statement timeouts
158k rows
```
Sequel::DatabaseError: PG::Error: ERROR: canceling statement due to statement timeout

File "services/table-geocoder/lib/table_geocoder.rb", line 134 in load_results_into_original_table
File "services/table-geocoder/lib/table_geocoder.rb", line 94 in process_results
File "app/models/geocoding.rb", line 119 in run_geocoding!
File "app/models/geocoding.rb", line 99 in run!
File "lib/resque/geocoder_jobs.rb", line 9 in block in perform
File "lib/resque/base_job.rb", line 21 in call
File "lib/resque/base_job.rb", line 21 in run_action
File "lib/resque/geocoder_jobs.rb", line 9 in perform
```

Relates to https://github.com/CartoDB/cartodb-platform/issues/423 (and needs that change to make sure sessions are renewed and PGBouncer doesn't keeps them)

### PR DESCRIPTION
**Pull request by Kartones**
Fixes #3143

### CODE REVIEW

**Comment by Kartones** [comment]
@rafatower  CR please

@javisantana & @zenitraM Check it, for now is what we have, but probably will need PGBouncer out or we cannot be sure connection will be dropped (I'm only dropping Rails pooler ones)

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2949/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2951/))

File: app/models/geocoding.rb [hunk 1]
```diff
@@ -54,13 +55,21 @@ def error
 
   def table_geocoder
     geocoder_class = (kind == 'high-resolution' ? CartoDB::TableGeocoder : CartoDB::InternalGeocoder::Geocoder)
+    # Reset old connections to make sure changes apply
+    if user.present?
+      user.reset_pooled_connections
+      user_connection = user.in_database(statement_timeout: DB_TIMEOUT)
```
Comment (rafatower) [inline app/models/geocoding.rb]: I'd wrap these two actions in a method and add a big comment stating that it should be called only from within a resque job.

**Comment by rafatower** [comment]
:+1:

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2968/))

**Pull request merged**